packages/graph/hash_graph/lib/graph/src/store/query/filter.rs

```diff
@@ -27,6 +31,47 @@ pub enum Filter<'q, T: QueryRecord> {
     ),
 }
 
+macro_rules! define_ontology_filters {
```

Comment:
I'm okay with this, @Alfred-Mountfield, do you have anything against this?
I think that especially since the `macro_rules` is right next to the usage and not scattered around that it makes it fine compared to the alternative of making this generic or 4x code duplication.

Reply:
It seems fairly contained, and very readable. I'm not against it, I'd prefer if we avoided macros where we can, but I also don't know if it's worth time looking at if we can do better here right now.